component/src/main/java/org/wso2/extension/siddhi/io/http/sink/HttpSink.java

```diff
@@ -561,13 +631,188 @@ protected void init(StreamDefinition outputStreamDefinition, OptionHolder option
      */
     @Override
     public void publish(Object payload, DynamicOptions dynamicOptions) {
-        if (!publisherURLOption.isStatic()) {
-            initClientConnector(dynamicOptions);
-        }
+        //get the dynamic parameter
         String headers = httpHeaderOption.getValue(dynamicOptions);
+        List<Header> headersList = HttpSinkUtil.getHeaders(headers);
+
+        if (authType.equals(HttpConstants.BASIC_AUTH) || authType.equals(HttpConstants.NO_AUTH)) {
+            sendRequest(payload, dynamicOptions, headersList);
+        } else {
+            sendOauthRequest(payload, dynamicOptions, headersList);
+        }
+    }
+
+    private void sendOauthRequest(Object payload, DynamicOptions dynamicOptions, List<Header> headersList) {
+        //generate encoded base64 auth for getting refresh token
+        String consumerKeyValue = consumerKey + ":" + consumerSecret;
+        String encodedRefreshAuth = "Basic " + encodeBase64(consumerKeyValue);
+        //check the availability of access token in the header
+        checkAccessToken(encodedRefreshAuth, dynamicOptions, headersList);
+        //send a request to API and get the response
+        int response = sendRequest(payload, dynamicOptions, headersList);
+        //if authentication fails then get the new access token
+        if (response == HttpConstants.AUTHENTICATION_FAIL_CODE) {
+            oauthFails(payload, dynamicOptions, headersList, encodedRefreshAuth);
+        } else if (response == HttpConstants.SUCCESS_CODE) {
+            log.info("Request send successfully.");
```

Comment:
```suggestion
            log.info("Request sent successfully.");
```

Also, please add more context information to make this log message useful to the user.  
IMO it is sufficient to have this as a debug log.